### ISSUE TITLE
tls: client offered an unsupported, maximum protocol version of 300

### ISSUE THREAD

**Issue by babolivier** [mholt/caddy#1241]
### 1. What version of Caddy are you running (`caddy -version`)?

Caddy 0.9.3

### 2. What are you trying to do?

Reverse proxy a [synapse](/matrix-org/synapse) server, and accessing it from the [Riot](https://f-droid.org/repository/browse/?fdfilter=riot&fdid=im.vector.alpha) Android app.

### 3. What is your entire Caddyfile?
```text
synapse.club-tech.fr {
	tls brendan.abolivier@isen-bretagne.fr
	proxy / 127.0.0.1:8008 {
		transparent
	}
	log matrix.access.log {
		import logrotate.caddyfile
	}
}
```

### 4. How did you run Caddy (give the full command and describe the execution environment)?

```
ExecStart=/usr/local/bin/caddy -log stdout -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp
```
Using the default systemd service.

### 5. What did you expect to see?

Connexion between the server and the client app going through Matrix's authentication.

### 6. What did you see instead (give full error messages and/or log)?

"Handshake failed" in the app, and

```
2016/11/06 19:34:43 http: TLS handshake error from [my phone's IP]:1107: remote error: tls: internal error
2016/11/06 19:34:43 http: TLS handshake error from [my phone's IP]:1260: tls: client offered an unsupported, maximum protocol version of 300
```

in the Caddy error log.

When accessing the server directly, everything works fine.

### 7. How can someone who is starting from scratch reproduce this behavior as minimally as possible?

Install Synapse, use the above Caddy configuration, and try to access it via the Riot Android app (using the hostname and port set up in Caddy so it goes through Caddy).

**Comment by mholt** [comment]
It looks like your app doesn't support modern TLS protocols. Can you reproduce this error without Synapse? And riot? Simplify your setup to reduce error surface area.

**Comment by babolivier** [comment]
I don't think it makes sense to reproduce without the app, as it seems to be the one producing the error, or Synapse because the request clearly don't go that far.

As far as I understood, the issue lies within the Riot app, so I opened an issue on their repo under https://github.com/vector-im/vector-android/issues/680

**Comment by mholt** [comment]
> I don't think it makes sense to reproduce without the app, as it seems to be the one producing the error

And that's what I was getting at -- thanks for opening the issue with the other project. :)

I'll close this for now but if some new evidence can be presented and this can be confirmed to be a bug in Caddy we will re-investigate it!

**Comment by sbuller** [comment]
This message has been making a mess of my log files, drowning anything else that /might/ be in there. It looks like in my case the message is caused by HAProxy's health checking. I've now changed HAProxy's health check method from SSL to Basic which will just check whether it can create a basic socket connection instead of looking for an SSL endpoint. I don't know if any of that makes a difference here, but I thought I'd post, since I came across this issue while figuring out what was going on.
